js/test/utils.test.js

```diff
@@ -109,15 +169,86 @@ describe('utils', () => {
   });
 
   describe('#safeSlice', () => {
-    it('returns a safe slice on an array');
-    // throws RangeError("BigInt argument out of safe number range") if `first` is out of range
-    // throws RangeError("BigInt argument out of safe number range") if `last` is out of range
-    // throws Error("Underflow during subtraction.") if `first` is less than 0 // unecessary???
-    // throws Error("Underflow during subtraction.") if `last` is less than 0 // unecessary???
-    // throws Error("Tried to slice past end of array") if `end` if larger than array length
-    // throws Error("Slice must not use negative indexes") if `start` is negative
-    // throws Error("Slice must not use negative indexes") if `end` is negative
-    // throws Error("Slice must not have 0 length") is `start` is greater than or equal to `end`
+    it('returns a safe slice on an array', () => {
+      const arr = new Uint8Array([1,2,3,4,5]);
+      let res;
+      let arraysAreEqual;
+
+      // regular slice
+      res = utils.safeSlice(arr, 0, 3);
+      arraysAreEqual = utils.typedArraysAreEqual(res, new Uint8Array([1,2,3]));
+      assert.isTrue(arraysAreEqual);
+
+      // slice that copies the original array
+      res = utils.safeSlice(arr);
+      arraysAreEqual = utils.typedArraysAreEqual(res, arr);
+      assert.isTrue(arraysAreEqual);
+
+      // slice with start index, but not end index
+      res = utils.safeSlice(arr, 2);
+      arraysAreEqual = utils.typedArraysAreEqual(res, new Uint8Array([3,4,5]));
+      assert.isTrue(arraysAreEqual);
+    });
+    it('error if passed invalid arguments', () => {
+      const arr = [1,2,3,4,5];
+      const OUT_OF_RANGE = BigInt(Number.MAX_SAFE_INTEGER + 1)
```

Comment:
`const OUT_OF_RANGE = BigInt(Number.MAX_SAFE_INTEGER) + BigInt(1);`

Reply:
Is this necessary to change?  Both ways return the same value.

Reply:
The original way wasn't passing when I ran the tests.

Reply:
"max safe integer" is what it sounds like. Above that the behavior is unsafe, which is why it wouldn't work consistently

Reply:
Oh, I understand now. Yes, you're right, I'll change this.